packages/jest-transform/src/ScriptTransformer.ts

```diff
@@ -175,6 +258,41 @@ export default class ScriptTransformer {
     return undefined;
   }
 
+  private async _getTransformerAsync(filename: Config.Path) {
+    let transform: Transformer | null = null;
+    if (!this._config.transform || !this._config.transform.length) {
+      return null;
+    }
+
+    const transformPath = this._getTransformPath(filename);
+    if (transformPath) {
+      const transformer = this._transformCache.get(transformPath);
+      if (transformer != null) {
+        return transformer;
+      }
+
+      transform = await import(transformPath);
```

Comment:
If this is to `import` ESM we need to add this gile to the blacklist in babel.config.js in the root of the repo.